web/client/reducers/widgets.js

```diff
@@ -158,9 +158,16 @@ function widgetsReducer(state = emptyState, action) {
                             // every chart stores the layer object configuration
                             // so we need to loop around them to update correctly the layer properties
                             // including the layerFilter
-                            return set("charts", w.charts.map((chart) =>
-                                get(chart, "layer.id") === action.layer.id ? set("layer", action.layer, chart) : chart
-                            ), w);
+                            let chartsCopy = w?.charts?.length ? [...w.charts] : [];
+                            chartsCopy = chartsCopy.map(chart=>{
+                                let chartItem = {...chart};
+                                chartItem = get(chart, "layer.id") === action.layer.id ? set("layer", action.layer, chart) : chart;
```

Comment:
this line is not needed. remove it please 
this is because now charts are using traces and layer has been moved inside traces

you can have a look at the schema here 
[charts-schema.zip](https://github.com/geosolutions-it/MapStore2/files/13659080/charts-schema.zip)

also fix tests accordingly